src/page/sinfo.rs

```diff
@@ -32,28 +32,39 @@ impl core::fmt::Display for SecInfo {
         match self.class {
             Class::Secs => write!(f, "S"),
             Class::Tcs => write!(f, "T"),
-            Class::Reg => write!(f, "{}", self.flags),
-            Class::Va => write!(f, "V"),
-            Class::Trim => write!(f, "^"),
+            Class::Regular => write!(f, "{}", self.flags),
+            Class::VersionArray => write!(f, "V"),
+            Class::Trimmed => write!(f, "^"),
+            Class::ShadowStackFirst => write!(f, "F"),
+            Class::ShadowStackRest => write!(f, "R"),
         }
     }
 }
 
-impl SecInfo {
-    /// Creates a `SecInfo` instance for regular pages
-    pub const fn reg(flags: Flags) -> Self {
-        Self {
-            flags,
-            class: Class::Reg,
-            reserved: [0; 31],
-        }
+impl From<Class> for SecInfo {
+    fn from(class: Class) -> Self {
+        SecInfo::new(class, None)
     }
+}
+
+impl SecInfo {
+    /// Create a new instance.
+    #[inline]
+    pub fn new(class: Class, flags: impl Into<Option<Flags>>) -> SecInfo {
+        let flags = match flags.into() {
+            Some(flags) => flags,
+            None => {
+                match class {
+                    // A CPU constraint
+                    Class::Regular => Flags::READ,
+                    _ => Flags::empty(),
+                }
+            }
+        };
```

Comment:
After this I would add
```rust
assert!(class != Class::Regular || (flags & Flags::READ) != 0));
```

Reply:
Rather than two match statements here, you should use `flags.into().unwrap_or_else(|| match class { .. })`.